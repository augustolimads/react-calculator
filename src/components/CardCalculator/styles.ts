import styled from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  background: inherit;
  width: 28em;
  ${media.lessThan('medium')`
  width: calc(100vw - 2.5em);
  `}
`

export const Head = styled.div`
  border: 1px solid red;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 6.4em;
  ${media.lessThan('medium')`
  height: 8em;
  `}
`

export const Body = styled.div`
  border: 1px solid red;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(5, 1fr);
`
export const Equal = styled.h2`
  grid-column: 4 / 4;
  grid-row-start: 4;
  grid-row-end: span 5;
  height: 100%;
  width: 100%;
`
