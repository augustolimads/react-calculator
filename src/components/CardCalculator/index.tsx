import * as S from './styles'

const CardCalculator = () => (
  <S.Wrapper>
    <S.Head>
      <h2>head</h2>
    </S.Head>
    <S.Body>
      <h2>C</h2>
      <h2>%</h2>
      <h2>/</h2>
      <h2>x</h2>
      <h2>7</h2>
      <h2>8</h2>
      <h2>9</h2>
      <h2>+</h2>
      <h2>4</h2>
      <h2>5</h2>
      <h2>6</h2>
      <h2>-</h2>
      <h2>1</h2>
      <h2>2</h2>
      <h2>3</h2>
      <h2>back</h2>
      <h2>0</h2>
      <h2>.</h2>
      <S.Equal>Equal</S.Equal>
    </S.Body>
  </S.Wrapper>
)

export default CardCalculator
