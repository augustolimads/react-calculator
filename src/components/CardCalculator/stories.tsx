import { Story, Meta } from '@storybook/react/types-6-0'
import Container from 'components/Container'
import CardCalculator from '.'

export default {
  title: 'CardCalculator',
  component: CardCalculator
} as Meta

export const Default: Story = () => (
  <Container>
    <CardCalculator />
  </Container>
)
