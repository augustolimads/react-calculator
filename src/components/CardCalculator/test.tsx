import { render, screen } from '@testing-library/react'

import CardCalculator from '.'

describe('<CardCalculator />', () => {
  it('should render the heading', () => {
    const { container } = render(<CardCalculator />)
    expect(
      screen.getByRole('heading', { name: /CardCalculator/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
