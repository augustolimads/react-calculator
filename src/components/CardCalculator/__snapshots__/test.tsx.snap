// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCalculator /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CardCalculator
  </h1>
</main>
`;
